app/sagas/session.js

```diff
@@ -13,14 +15,48 @@ export function* autoJoinUser(action) {
             sessionId,
             user: currentUser
         }});
+        storeSessionToLocalStorage(currentUser, sessionId);
     }
 }
 
+export function* renameCurrentSessionInLocalStorage (action) {
+    console.log("renaming session");
+    console.log(action);
+    const currentSession = yield select(state => state.session.id);
+    const currentUser = yield select(state => state.user.name);
+    let savedSessions = ls.get('sessions');
+    let savedSession  = find(savedSessions[currentUser], session => session.id === currentSession);
+    savedSession.name = action.payload;
+    ls.set('sessions', savedSessions);
+}
+
+function storeSessionToLocalStorage(currentUser, sessionId) {
+    let savedSessions = ls.get('sessions');
```

Comment:
The reason I made it like this is because you give the users the option to log out. Therefor I assumed multiple users could be using the same browser.